design/cstor/v1/pool_expansion_by_expanding_disk.md

```diff
@@ -0,0 +1,158 @@
+
+Below are high level info to explain the Day2 operations that are currently supported.
+
+- Pool Expansion When BlockDevice/raid group added:
+	cstor-pool-mgmt will iterate over all the BlockDevices that exists in CStorPoolInstance resource and verify whether the corresponding BlockDevice/raid group is part of the pool. If that BlockDevice/raid group is not part of the pool then pool expansion will be triggered on newly added BlockDevice/raid group.
+
+- BlockDevice Replacement In Pool:
+    cstor-pool manager will iterate over all the BlockDevices present in CstorPoolInstance and verifies the requested BlockDevice replacement request (we are maintaining annotation called openebs.io/predecessor: <replaced_blockdevice_name> on the claim of that blockdevice). If requested BlockDevice is valid for replacement then the pool managet will perform BlockDevice replacement operation on the pool.
+
+### Proposed Implementations
+
+- CStor-pool Manger inbuilt CSPI controller watches for CSPI resources and tries to get the desired state via every sync operation. It will iterate over all the BlockDevices that exist on CStorPoolInstance resource and if BlockDevice resource capacity(considerable changes) is greater than the existing capacity of BlockDevice on CstorPoolInstanceBlockDevice then we can follow below-mentioned steps to perform the pool expansion:
+1. If IsBlockDeviceExpanded flag sets true then it is something like user/admin is requesting to perform pool expansion steps.
+2. If IsBlockDeviceExpanded flag sets false then NoOps will be performed.
+
+Currently CstorPoolInstanceBlockDevice holds below configurations 
+```go
+// CStorPoolInstanceBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolInstanceBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
+}
+```
+
+To get user Input proposed to add field in above struct
+
+```go
+// CStorPoolInstanceBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolInstanceBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
+
+	// IsBlockDeviceExpanded informs whether blockdevice 
+	// got expanded or not
+	IsBlockDeviceExpanded bool `json:”isBlockDeviceExpanded”`
+}
+```
+
+### Steps to perform user stories
+- User has to set the value of IsBlockDeviceExpanded to true in corresponding CStorPoolInstanceBlockDevice.
```

Comment:
when is this supposed to be unset?

Reply:
Users can unset it once the operation is completed by looking at events/increased capacity. Leaving As is also fine(No problem).

Reply:
I will update this point in PR.